src/cache.jl

```diff
@@ -0,0 +1,103 @@
+if VERSION <= v"0.6.0-dev.2474"
+    import Base: subtypes
+    subtypes(m::Module, x::DataType) = x.abstract ? sort!(collect(_subtypes(m, x)), by=string) : DataType[]
+end
+
+function modnames(m::AbstractString, top)
+    s = Symbol(m)
+    eval(:(using $s))
+    M, exists = Base.REPLCompletions.get_value(s, Main)
+    if !(s in keys(top))
+        modnames(M, top)
+    end
+end
+
+function modnames(M::Module, top)
+    s = parse(string(M))
+    d = Dict{Any,Any}(:EXPORTEDNAMES=>setdiff(names(M), [:Function]))
+    top[s] = d
+    for n in names(M, true, true)
+        if !Base.isdeprecated(M, n) && first(string(n))!="#" && isdefined(M, n) && n!=:Function
+            x = eval(M, n)
+            if isa(x, Module) && x!=M
+                s = parse(string(x))
+                if s in keys(top)
+                    d[n] = top[s]
+                else
+                    d[n] = modnames(x, top)
+                end
+            elseif first(string(n))!='#'
+                t = Symbol(isa(x, Function) ? Function :
+                           isa(x, DataType) ? DataType :
+                                              typeof(x))
+                if isa(x, DataType) && x.abstract
+                    doc = "$n <: $(x.super)"
+                else
+                    doc = string(Docs.doc(Docs.Binding(M, n)))
+                end
+                d[n] = (t, doc, sig(x))
+            end
+        end
+    end
+    return d
+end
+
+sig(x) = []
+function sig(x::Union{DataType,Function})
+    out = []
+    for m in methods(x)
+        p = string.(collect(m.sig.parameters[2:end]))
+        push!(out, (String(m.file), m.line, m.source.slotnames[2:length(p)+1], p))
```

Comment:
This line crashes for me. I can get rid of the first crash by using ``string(m.file)`` instead of ``String(m.file)``, but then I get another crash
````
ERROR: type Method has no field source
 in sig(::Function) at C:\Users\anthoff\.julia\v0.5\LanguageServer\src\cache.jl:50
 in modnames(::Module, ::Dict{Any,Any}) at C:\Users\anthoff\.julia\v0.5\LanguageServer\src\cache.jl:38
 in eval(::Module, ::Any) at .\boot.jl:234
 in process_options(::Base.JLOptions) at .\client.jl:239
 in _start() at .\client.jl:318
````
I guess this is some julia 0.5 vs 0.6 error?